Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -91,97 +216,72 @@ CGColorRef CGColorCreateCopy(CGColorRef color) {
-    [static_cast<UIColor*>(ret) setColorSpace:CGColorGetColorSpace(color)];
-    return ret;
-}
-
-/**
- @Status Caveat
- @Notes components are not supported
-*/
-CGColorRef CGColorCreateWithPattern(CGColorSpaceRef colorSpace, CGPatternRef pattern, const CGFloat* components) {
-    RETURN_NULL_IF(!colorSpace);
-    RETURN_NULL_IF(!pattern);
-    CGColorRef ret = static_cast<CGColorRef>([[__LazyUIColor _colorWithCGPattern:(CGPatternRef)pattern] retain]);
-    [static_cast<UIColor*>(ret) setColorSpace:colorSpace];
-    return ret;
+    return __CGColor::CreateInstance(kCFAllocatorDefault, color, alpha);
 }
 
 /**
  @Status Interoperable
 */
 bool CGColorEqualToColor(CGColorRef color1, CGColorRef color2) {
-    if (color1 == color2) {
-        return true;
-    }
-    if (!color1 || !color2) {
-        return false;
-    }
-    const __CGColorQuad* components1 = [(UIColor*)color1 _getColors];
-    const __CGColorQuad* components2 = [(UIColor*)color2 _getColors];
-
-    return ((components1->r == components2->r) && (components1->g == components2->g) && (components1->b == components2->b) &&
-            (components1->a == components2->a)) &&
-           (CGColorSpaceGetModel(CGColorGetColorSpace(color1)) == CGColorSpaceGetModel(CGColorGetColorSpace(color2)));
+    RETURN_RESULT_IF(color1 == color2, true);
+    RETURN_FALSE_IF(!color1 || !color2);
+    return CFEqual(color1, color2);
 }
 
 /**
  @Status Interoperable
 */
 CGFloat CGColorGetAlpha(CGColorRef color) {
     RETURN_RESULT_IF(!color, 0.0f);
-    const __CGColorQuad* curColor = [(UIColor*)color _getColors];
-    return curColor->a;
+    return color->Alpha();
 }
 
 /**
  @Status Interoperable
 */
 CGColorSpaceRef CGColorGetColorSpace(CGColorRef color) {
     RETURN_NULL_IF(!color);
-    return [static_cast<UIColor*>(color) colorSpace];
+    return color->ColorSpace();
 }
 
 /**
  @Status Caveat
  @Notes Works as expected for RGBA only.
 */
 const CGFloat* CGColorGetComponents(CGColorRef color) {
-    return (const CGFloat*)[(UIColor*)color _getColors];
+    RETURN_NULL_IF(!color);
+    return reinterpret_cast<const CGFloat*>(&color->ColorComponents());
 }
 
 /**
  @Status Caveat
  @Notes Limited support for colorspace
 */
 size_t CGColorGetNumberOfComponents(CGColorRef color) {
-    return CGColorSpaceGetNumberOfComponents(CGColorGetColorSpace(color)) + 1;
+    return CGColorSpaceGetNumberOfComponents(color->ColorSpace()) + 1;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 CGPatternRef CGColorGetPattern(CGColorRef color) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    RETURN_NULL_IF(!color);
+    return color->Pattern();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 CFTypeID CGColorGetTypeID() {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return __CGColor::GetTypeID();
+}
+
+#pragma region InternalHelpers
+
+CGColorRef _CGColorCreateWithComponents(CGColorSpaceRef colorspace, CGFloat r, CGFloat g, CGFloat b, CGFloat alpha) {
```

Comment:
`CGColorCreateGenericRGB`  now how did I forget about that :(
this will allow us to add the missing apis